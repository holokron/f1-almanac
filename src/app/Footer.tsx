import Image from "next/image";
import LogoImage from "~/../public/logo.png";

export default function Footer(): JSX.Element {
  return (
    <footer className="footer bg-neutral p-10 text-neutral-content">
      <div>
        <Image src={LogoImage} alt="logo" width={64} />
        <p>F1 Almanac</p>
      </div>
      <div>
        <span className="footer-title">Built using</span>
        <a
          className="link-hover link"
          href="https://react.dev/"
          target="_blank"
          rel="nofollow"
        >
          React
        </a>

        <a
          className="link-hover link"
          href="https://nextjs.org/"
          target="_blank"
          rel="nofollow"
        >
          Next.js
        </a>

        <a
          className="link-hover link"
          href="https://trpc.io/"
          target="_blank"
          rel="nofollow"
        >
          tRPC
        </a>

        <a
          className="link-hover link"
          href="https://tailwindcss.com/"
          target="_blank"
          rel="nofollow"
        >
          Tailwind CSS
        </a>

        <a
          className="link-hover link"
          href="https://create.t3.gg/"
          target="_blank"
          rel="nofollow"
        >
          Create T3 App
        </a>
      </div>

      <div>
        <span className="footer-title">Data sources</span>
        <a
          className="link-hover link"
          href="http://ergast.com/mrd/"
          target="_blank"
          rel="nofollow"
        >
          Ergast Developer API
        </a>

        <a
          className="link-hover link"
          href="https://www.wikipedia.org/"
          target="_blank"
          rel="nofollow"
        >
          Wikipedia
        </a>
      </div>
    </footer>
  );
}
