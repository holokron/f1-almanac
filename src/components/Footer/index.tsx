import React from "react"
import { Row, Typography } from "antd"

export default React.memo(
  (): React.ReactElement => (
    <Row type="flex" align="middle" justify="center" color="green-1">
      <Typography.Text>
        v0.1.0 &copy; 2019&nbsp;
        <a href="https://github.com/holokron">Holokron</a>
      </Typography.Text>
    </Row>
  ),
)
