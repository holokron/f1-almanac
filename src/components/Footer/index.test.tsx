import React from "react"
import renderer from "react-test-renderer"
import Footer from "."

describe("components/Footer", (): void => {
  it.each([["2019", "1.2.3"]])(
    "data: %p",
    (copyright: string, version: string): void => {
      const tree = renderer.create(
        <Footer copyright={copyright} version={version} />,
      )

      expect(tree.toJSON()).toMatchSnapshot()
    },
  )
})
