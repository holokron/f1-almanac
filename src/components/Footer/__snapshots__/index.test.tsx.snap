// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Footer data: "2019" 1`] = `
<div
  className="ant-row-flex ant-row-flex-center ant-row-flex-middle"
  color="green-1"
>
  <span
    aria-label={null}
    className="ant-typography"
    style={
      Object {
        "WebkitLineClamp": null,
      }
    }
  >
    v
    1.2.3
     © 
    2019
     
    <a
      href="https://github.com/holokron"
    >
      Holokron
    </a>
  </span>
</div>
`;
